torch/_inductor/select_algorithm.py

```diff
@@ -719,6 +756,41 @@ def benchmark(choice):
             torch.cuda.synchronize()  # shake out any CUDA errors
             return min(result)
 
+        def benchmark_in_sub_process(choice):
+            out.zero_()
+
+            if isinstance(choice, ExternKernelCaller):
+                inputs = example_inputs_extern
+                output = out_extern
+            else:
+                inputs = example_inputs
+                output = out
+
+            if VERIFY:
+                expected_output = expected
+            else:
+                expected_output = None
+
+            # use a tensor since the mutation to a python list in a sub process
+            # is not synced back to the parent process
+            timings = torch.zeros(3, dtype=torch.float32)
+
+            child = multiprocessing.Process(target=benchmark_choice_in_sub_process, args=(dumps(template_kernels), dumps(choice), inputs, output, expected_output, timings))
+            child.start()
+            child.join()
+
+            # child process fail
+            if child.exitcode != 0:
+                warnings.warn(f"Fail to benchmark choice '{choice}'. It will be ignored. Please debug the root cause in case the choice can bring perf gains.")
+
+                # return a large value to this choice will be ignored
+                return 1e10
+
+            torch.cuda.synchronize()  # shake out any CUDA errors
+            return timings.min().item()
```

Comment:
it's not too important, but `do_bench` returns 20-th, 50-th and 80-th percentile, when we are getting `timings.min()` we are getting 20-th, and we probably want 50-th? Same issue in regular single-process benchmark

Reply:
yea, I literally copied the logic from single-process case :)  I think it make sense to use 50-th percentile / median. That's also what we do in triton_ops/autotune.py . I'll update this.